server/connectors/slack_connector/client.py

```
@@ -39,6 +44,35 @@ def _get_retry_delay(attempt: int, response=None) -> int:
+        fall back to time.sleep().
+        """
+        try:
+            loop = asyncio.get_event_loop()
+        except RuntimeError:
+            loop = None
+
+        if loop is not None and loop.is_running():
+            future = asyncio.run_coroutine_threadsafe(asyncio.sleep(seconds), loop)
+            try:
+                future.result(timeout=seconds + 5)
+            except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)